src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -630,20 +631,32 @@ static String altKey(JsonObject jsonObject, String... keys) {
     if (counterReportTitle == null) {
       return Future.succeededFuture();
     }
+    String yop = reportItem.getString("YOP");
     final String usageDateRange = getUsageDate(reportItem);
     final JsonObject identifiers = getIssnIdentifiers(reportItem);
     final String onlineIssn = identifiers.getString("onlineISSN");
     final String printIssn = identifiers.getString("printISSN");
     final String isbn = identifiers.getString("ISBN");
     final String doi = identifiers.getString("DOI");
-    final String publicationDate = identifiers.getString("Publication_Date");
+    String publicationDate = identifiers.getString("Publication_Date");
+
+    DateTimeFormatter format = new DateTimeFormatterBuilder()
+        .appendPattern("yyyy")
+        .parseDefaulting(ChronoField.MONTH_OF_YEAR, 1)
+        .parseDefaulting(ChronoField.DAY_OF_MONTH, 1)
+        .toFormatter();
+
+    final LocalDate pubdate = publicationDate != null
+        ? LocalDate.parse(publicationDate) : yop != null
+        ? LocalDate.parse(yop, format) : null;
```

Comment:
This is more easy to understand if the conditions are at the beginning of the line, see example
https://en.wikipedia.org/wiki/%3F:#Case_selectors

Reply:
ok. Done